cmd/snap/cmd_list.go

```diff
@@ -124,6 +131,60 @@ func listSnaps(names []string, all bool) error {
 	w := tabWriter()
 	defer w.Flush()
 
+	switch format {
+	case "":
+		return outputSnapsDefault(w, snaps)
+	default:
+		return outputSnapsWithFormat(Stdout, snaps, format)
+	}
+}
+
+func clientSnapFields() []string {
+	v := reflect.ValueOf(client.Snap{})
+	n := v.Type().NumField()
+	fields := make([]string, n)
+	for i := 0; i < n; i++ {
+		fields[i] = v.Type().Field(i).Name
+	}
+
+	sort.Strings(fields)
+	return fields
+}
+
+func describeListFormat(w io.Writer) error {
+	fmt.Fprintf(w, `Format uses a simple template system.
+
+Use --format="{{.Name}} {{.Version}}" to get started.
+
+All elements available for snaps are:
+`)
+	for _, fld := range clientSnapFields() {
+		fmt.Fprintf(w, " - %s\n", fld)
```

Comment:
it would be nice if `clientSnapFields` returned things that had the name and a short help.
Something like http://paste.ubuntu.com/p/4cNWwKhtFP/

Reply:
Nice one! I updated the code accordingly. I wonder if we should omit showing fields that have no "help" ? E.g. the `.Icon` field is probably not super useful for most people.